src/main/kotlin/com/campspot/middleware/transactions/TransactionAspect.kt

```diff
@@ -0,0 +1,45 @@
+package com.campspot.middleware.transactions
+
+import org.jdbi.v3.core.Handle
+import org.jdbi.v3.core.Jdbi
+
+class TransactionAspect(private val dbis: Map<String, Jdbi>, private val daoManager: DAOManager) {
+  private var inTransaction: InTransaction? = null
+  private var handle: Handle? = null
+
+  fun beforeStart(inTransaction: InTransaction) {
+    this.inTransaction = inTransaction
+    val jdbi = dbis[inTransaction.name]
+    handle = jdbi?.open()
```

Comment:
This should probably use `!!`, that way it will blow up here if someone uses the wrong name for the connection they want. Otherwise it will just blow up when they try to actually run a query